_posts/2016-04-14-Rust-1.8.md

```diff
@@ -0,0 +1,251 @@
+---
+layout: post
+title: "Announcing Rust 1.8"
+author: The Rust Core Team
+---
+
+The Rust team is happy to announce the latest version of Rust, 1.8. Rust is a
+systems programming language focused on safety, speed, and concurrency.
+
+As always, you can [install Rust 1.8][install] from the appropriate page on our
+website, and check out the [detailed release notes for 1.8][notes] on GitHub.
+About 1400 patches were landed in this release.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/stable/RELEASES.md#version-180-2016-04-14
+
+### What's in 1.8 stable
+
+There are two new features in Rust 1.8, as well as good news for Windows users!
+Additionally, work is underway to replace our `make`-based build system with
+one based on Cargo.
+
+The first feature is that the various “operator equals” operators, such as `+=`
+and `-=`, are now overloadable via various traits. This change was accepted in
+[RFC 953], and looks like this:
+
+```rust
+use std::ops:: AddAssign;
+
+#[derive(Debug)]
+struct Count { 
+    value: i32,
+}
+    
+impl AddAssign for Count {
+    fn add_assign(&mut self, other: Count) {
+        self.value += other.value;
+    }
+}   
+
+fn main() {
+    let mut c1 = Count { value: 1 };
+    let c2 = Count { value: 5 };
+
+    c1 += c2;
+
+    println!("{:?}", c1);
+}
+```
+
+[RFC 953]: https://github.com/rust-lang/rfcs/blob/master/text/0953-op-assign.md
+
+This will print out `Count { value: 6 }`. Like the other operator traits, an
+associated type allows you to use different types on each side of the operator,
+as well. See the RFC for more details.
+
+The second feature is very small, and comes from [RFC 218]. Before Rust 1.8, a
+`struct` with no fields did not have curly braces:
+
+```rust
+struct Foo; // works
+struct Bar { }; // error
+```
+
+[RFC 218]: https://github.com/rust-lang/rfcs/blob/master/text/0218-empty-struct-with-braces.md
+
+The second form is no longer an error, and is equivalent to the first. This was
+originally disallowed for consistency with other empty declarations, as well as
+a parsing ambiguity. However, that ambiguity is non-existent in post-1.0 Rust,
+and macro authors saw additional complexity due to needing a special-case. Also,
+users who do active development would sometimes switch between empty and
+non-empty versions of a struct, and the extra work and diffs involved was less
+than ideal. Allowing both forms is slightly less consistent, but the ergonomics
+of these two cases was deemed worth it.
```

Comment:
I feel like this is a bit more defensive than necessary. From my point-of-view, permitting `{ }` is **more** consistent, not less. That is, `struct Foo;` is a kind of shorthand for the combination of `struct Foo { }` and `const Foo: Foo = Foo { };`. The fact that we only permitted the shorthand before was a bit odd.